double free or corruption (!prev) error on exit (GDNative c++)

**Godot version:**
Godot Engine v3.3.rc7.official

**OS/device including version:**
Linux Ubuntu 20.04

**Issue description:**
GDnative C++ application runs fine, but throws error in console: `double free or corruption (!prev)` on normal application exit.
Not visible when run from editor.

**Steps to reproduce:**
Any GDnative C++ project, start and exit.

**Minimal reproduction project:**
Any project, like this one https://github.com/BastiaanOlij/gdnative_cpp_example
Error happens somewhere after `Godot::gdnative_terminate(o);`

Hey guys, trying to help with debugging the issue, but I can't even get the godot-cpp sample https://github.com/BastiaanOlij/gdnative_cpp_example to compile on Windows. I get errors in `Object.hpp`.  
```
include\gen\Object.hpp(70): error C2143: syntax error: missing ';' before '<'
include\gen\Object.hpp(70): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
include\gen\Object.hpp(70): error C2238: unexpected token(s) preceding ';'
include\gen\Object.hpp(90): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
include\gen\Object.hpp(90): error C2143: syntax error: missing ',' before '<'
```

Not sure if it helps, but one thing I did notice was that the merge in #46844 removed the `library_classes.erase` statement inside of the `if(L)`.  I wonder if adding that back would make a difference.

--- a/modules/gdnative/nativescript/nativescript.cpp
+++ b/modules/gdnative/nativescript/nativescript.cpp
@@ -1759,6 +1759,8 @@ void NativeScriptLanguage::unregister_script(NativeScript *script) {
 						C->get().destroy_func.free_func(C->get().destroy_func.method_data);
 					}
 				}
+
+				library_classes.erase(script->lib_path);
 			}
 
 			Map<String, Ref<GDNative>>::Element *G = library_gdnatives.find(script->lib_path);

